File: juliapython/lorenz.jl
function rhs(dx,dy,dz,x,y,z)
    dx = σ*(y-x)
    dy = x*(ρ-z) - y
    dz = x*y - β*z
    return dx,dy,dz
end

function time_integration(N,x,y,z)
    dx,dy,dz = zeros((x,y,z))

    for i = 1:N
        dx,dy,dz = rhs(dx,dy,dz,x,y,z)
        x += dt*dx
        y += dt*dy
        z += dt*dz
    end
    return x,y,z
end

N = Int(1e8)
x,y,z = 5.,5.,5.

const σ = 10.
const ρ = 28.
const β = 8./3.

const dt = 0.01

time_integration(1,x,y,z)

@time x,y,z = time_integration(N,x,y,z)
println((x,y,z))

File: juliapython/lorenz.py
from time import time as tictoc

def time_integration(N,x,y,z):
    
    # pre-allocate
    dx,dy,dz = 0.,0.,0.
    
    for i in range(N-1):
        
        # RHS
        dx = dt*(sig*(y-x))
        dy = dt*(x*(rho-z) - y)
        dz = dt*(x*y - beta*z)
        
        # Euler forward
        x += dt*dx
        y += dt*dy
        z += dt*dz
    
    return x,y,z


N = int(1e8)        # number of time steps
dt = 0.01   # time step

# initial conditions
x,y,z = 5.,5.,5.    

# L63 parameters
sig = 10.
rho = 28.
beta = 8./3.


# RUN
t0 = tictoc()
x,y,z = time_integration(N,x,y,z)
t1 = tictoc()-t0
print("time elapsed {:.2f}s".format(t1))

print((x,y,z))

